File: Source/Bifrost.JavaScript.Specs/utils/for_StringMapping/when_resolving_a_string_with_only_a_wildcard.js
describe("when resolving a string with only a wildcard", function () {
    var expectedResult = "this.is.a.wildcard.uri.for_things";
    var mapping = Bifrost.utils.StringMapping.create({
        format: "**/",
        mappedFormat: "**."
    });
    var result = mapping.resolve("this/is/a/wildcard/uri/for_things");

    it("should expand input uri to the mapped uri", function () {
        expect(result).toBe(expectedResult);
    });
});

File: Source/Bifrost.Specs/Utils/for_StringMapping/when_resolving_a_string_with_only_a_wildcard.cs
using Bifrost.Utils;
using Machine.Specifications;

namespace Bifrost.Specs.Utils.for_StringMapping
{
    public class when_resolving_a_string_with_only_a_wildcard
    {
        const string expected_result = "this.is.a.wildcard.uri.for_things";

        static StringMapping mapping = new StringMapping(
                "**/",
                "**."
            );
        static string result;

        Because of = () => result = mapping.Resolve("this/is/a/wildcard/uri/for_things");

        It should_expand_uri_to_the_mapped_uri = () => result.ShouldEqual(expected_result);
    }
}
